projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -303,12 +303,10 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
         const searchAllBtn = this.esf.listData[0];
 
         if (!this.searchValue) {
-            const anyFiltered = this.esf.listData.some(i => i.isFiltered);
-            const anyUnfiltered = this.esf.listData.some(i => !i.isFiltered);
+            const listData = this.esf.listData.slice(1);
```

Comment:
Is the actual fix here removing the select all item? If that is the case I would still use the `anyFiltered`/`anyUnfiltered` approach on the sliced array since it is faster than filtering the whole collection.

Reply:
The fix actually is to always update the indeterminate state of the `Select All` checkbox, not setting it to true only when both anyFiltered and anyUnfiltered are true. Otherwise, the state that used to be indeterminate, stays so even when on subsequent opening all items are actually selected.